wagtail/documents/rich_text.py

```diff
@@ -31,7 +31,9 @@ def expand_db_attributes(attrs):
             doc = Document.objects.get(id=attrs['id'])
             return '<a data-linktype="document" data-id="%d" href="%s">' % (doc.id, escape(doc.url))
         except Document.DoesNotExist:
-            return "<a>"
+            return '<a data-linktype="document" data-id="%s">' % attrs['id']
```

Comment:
It might be worth adding a comment here. Outside of this PR, I would be very puzzled as why we print out the document's ID in the output despite the document not being found.